### PR TITLE
Add application tests

### PR DESCRIPTION
**Pull request by marmichalski**

### CODE REVIEW

File: tests/e2e/LicenseCheckerTest.php [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    }
+
+    public function test_failure_with_non_existent_allowed_licenses_file(): void
+    {
+        $exitCode = $this->commandTester->execute(['--allow-file' => 'dont_have.pls']);
+        $output = $this->getOutputLines();
+
+        self::assertSame(1, $exitCode);
+        self::assertCount(4, $output);
+        self::assertSame('[ERROR] File "dont_have.pls" could not be read.', $output[3]);
```
Comment (mannion007) [inline tests/e2e/LicenseCheckerTest.php:45]: Probably a bit fragile, why are there 4 lines, why is that message specifically at index 3 etc?

Comment (mannion007) [reply tests/e2e/LicenseCheckerTest.php:45]: (probably all that we care about is that it failed and for the expected reason, perhaps we can get that confidence with some less fragile assertions)

Comment (marmichalski) [reply tests/e2e/LicenseCheckerTest.php:45]:
```
.array(4) {
  [0]=>
  string(24) "Composer License Checker"
  [1]=>
  string(24) "========================"
  [2]=>
  string(0) ""
  [3]=>
  string(47) "[ERROR] File "dont_have.pls" could not be read."
}
```

:shrug:

Comment (mannion007) [reply tests/e2e/LicenseCheckerTest.php:45]: I mean, you could array filter over and just check the message you want it somewhere in the output, plus the correct exit code. Why do I care that it's at index 3?

Comment (marmichalski) [reply tests/e2e/LicenseCheckerTest.php:45]: We care that the output is an expected output, nothing more, nothing less. We don't care about the header (title?) part, which is always first 3 lines... but maybe we should also care about that? :see_no_evil:

File: tests/e2e/LicenseCheckerTest.php [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        $output = $this->getOutputLines();
+
+        self::assertSame(0, $exitCode);
+        self::assertCount(4, $output);
+        self::assertSame('[OK] All dependencies have allowed licenses.', $output[3]);
+    }
+
+    private function getOutputLines(): array
+    {
+        return \array_map('trim', \explode(\PHP_EOL, \trim($this->commandTester->getDisplay())));
```
Comment (ben-challis) [inline tests/e2e/LicenseCheckerTest.php:170]:
```suggestion
        return \array_map(\trim(...), \explode(\PHP_EOL, \trim($this->commandTester->getDisplay())));
```

File: tests/e2e/LicenseCheckerTest.php (hunk 1 above)
Comment (ben-challis) [reply tests/e2e/LicenseCheckerTest.php:45]: nit: maybe just compare it against a plain text expected outcome fixture file?

**Comment by ben-challis** [comment]
Let's merge this and improve down the line

**Review by ben-challis — APPROVED** [review]

**Pull request merged**